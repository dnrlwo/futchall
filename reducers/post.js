import produce from '../util/produce';

export const initialState = {
  mainPosts: [],
  imagePaths: [],
  post: null,
  selectPostLoading: false,
  selectPostDone: false,
  selectPostError: null,
  hasMorePosts: true,
  loadPostsLoading: false,
  loadPostsDone: false,
  loadPostsError: null,
  addPostLoading: false,
  addPostDone: false,
  addPostError: null,
  removePostLoading: false,
  removePostDone: false,
  removePostError: null,
  addCommentLoading: false,
  addCommentDone: false,
  addCommentError: null,
  uploadImagesLoading: false,
  uploadImagesDone: false,
  uploadImagesError: null,
};

export const UPLOAD_IMAGES_REQUEST = 'UPLOAD_IMAGES_REQUEST';
export const UPLOAD_IMAGES_SUCCESS = 'UPLOAD_IMAGES_SUCCESS';
export const UPLOAD_IMAGES_FAILURE = 'UPLOAD_IMAGES_FAILURE';

export const LOAD_POSTS_REQUEST = 'LOAD_POSTS_REQUEST';
export const LOAD_POSTS_SUCCESS = 'LOAD_POSTS_SUCCESS';
export const LOAD_POSTS_FAILURE = 'LOAD_POSTS_FAILURE';

export const ADD_POST_REQUEST = 'ADD_POST_REQUEST';
export const ADD_POST_SUCCESS = 'ADD_POST_SUCCESS';
export const ADD_POST_FAILURE = 'ADD_POST_FAILURE';

export const SELECT_POST_REQUEST = 'SELECT_POST_REQUEST';
export const SELECT_POST_SUCCESS = 'SELECT_POST_SUCCESS';
export const SELECT_POST_FAILURE = 'SELECT_POST_FAILURE';

export const REMOVE_POST_REQUEST = 'REMOVE_POST_REQUEST';
export const REMOVE_POST_SUCCESS = 'REMOVE_POST_SUCCESS';
export const REMOVE_POST_FAILURE = 'REMOVE_POST_FAILURE';

export const ADD_COMMENT_REQUEST = 'ADD_COMMENT_REQUEST';
export const ADD_COMMENT_SUCCESS = 'ADD_COMMENT_SUCCESS';
export const ADD_COMMENT_FAILURE = 'ADD_COMMENT_FAILURE';

export const CLEAR_ADD_COMMENT_RESULT = 'CLEAR_ADD_COMMENT_RESULT';

// 이전 상태를 액션을 통해 다음 상태로 만들어내는 함수(불변성은 지키면서)
const postReducer = (state = initialState, action) => produce(state, (draft) => {
  switch (action.type) {
    case LOAD_POSTS_REQUEST:
      draft.loadPostsLoading = true;
      draft.loadPostsDone = false;
      draft.loadPostsError = null;
      break;
    case LOAD_POSTS_SUCCESS:
      draft.loadPostsLoading = false;
      draft.loadPostsDone = true;
      draft.mainPosts = draft.mainPosts.concat(action.data);
      draft.hasMorePosts = action.data.length === 5;
      break;
    case LOAD_POSTS_FAILURE:
      draft.loadPostsLoading = false;
      draft.loadPostsError = action.error;
      break;
    case ADD_POST_REQUEST:
      draft.addPostLoading = true;
      draft.addPostDone = false;
      draft.addPostError = null;
      break;
    case ADD_POST_SUCCESS:
      draft.addPostLoading = false;
      draft.addPostDone = true;
      draft.mainPosts.unshift(action.data);
      break;
    case ADD_POST_FAILURE:
      draft.addPostLoading = false;
      draft.addPostError = action.error;
      break;
    case REMOVE_POST_REQUEST:
      draft.removePostLoading = true;
      draft.removePostDone = false;
      draft.removePostError = null;
      break;
    case REMOVE_POST_SUCCESS:
      draft.removePostLoading = false;
      draft.removePostDone = true;
      draft.mainPosts = draft.mainPosts.filter((v) => v.id !== action.data);
      break;
    case REMOVE_POST_FAILURE:
      draft.removePostLoading = false;
      draft.removePostError = action.error;
      break;
    case ADD_COMMENT_REQUEST:
      draft.addCommentLoading = true;
      draft.addCommentDone = false;
      draft.addCommentError = null;
      break;
    case ADD_COMMENT_SUCCESS: {
      const post = draft.mainPosts.find((v) => v.id === parseInt(action.data[0].PostId, 10));
      post.Comments = action.data;
      draft.addCommentLoading = false;
      draft.addCommentDone = true;
      break;
    }
    case ADD_COMMENT_FAILURE:
      draft.addCommentLoading = false;
      draft.addCommentError = action.error;
      break;
    case SELECT_POST_REQUEST:
      draft.selectPostLoading = true;
      draft.selectPostDone = false;
      draft.selectPostError = null;
      break;
    case SELECT_POST_SUCCESS:
      draft.post = action.data;
      draft.selectPostLoading = false;
      draft.selectPostDone = true;
      break;
    case SELECT_POST_FAILURE:
      draft.selectPostLoading = false;
      draft.selectPostError = action.error;
      break;
    case UPLOAD_IMAGES_REQUEST:
      draft.uploadImagesLoading = true;
      draft.uploadImagesDone = false;
      draft.uploadImagesError = null;
      break;
    case UPLOAD_IMAGES_SUCCESS:
      draft.imagePaths = action.data;
      draft.uploadImagesLoading = false;
      draft.uploadImagesDone = true;
      break;
    case UPLOAD_IMAGES_FAILURE:
      draft.uploadImagesLoading = false;
      draft.uploadImagesError = action.error;
      break;
    case CLEAR_ADD_COMMENT_RESULT:
      draft.addCommentDone = false;
      break;
    default:
      break;
  }
});

export default postReducer;
