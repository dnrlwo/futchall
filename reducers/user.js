import produce from '../util/produce';

export const initialState = {
  me: null, // 내 정보
  isLoggedIn: false, // 로그인 여부
  isLoggingOut: false, // 로그아웃 시도중
  isLoggingIn: false, // 로그인 시도중
  logInErrorReason: '', // 로그인 실패 사유
  isSignedUp: false, // 회원가입 성공
  isSigningUp: false, // 회원가입 시도중
  signUpErrorReason: '', // 회원가입 실패 사유
  isPreempting: false, // 회원가입시 아이디 중복체크중
  isPreempted: false, // 회원가입시 아이디 중복체크 완료
  preemptErrorReason: '', // 회원가입시 아이디 중복체크 실패 사유
  isChangingTo: false, // 회원정보 수정중
  isChangedTo: false, // 회원정보 수정완료
  changedToErrorReason: '', // 회원정보 수정 실패 사유
};

const dummyUser = {
  nickname: '우기재',
  userId: 'everest88',
  positions: ['PIVO', 'ALA', 'FIXO'],
  age: '20',
  locations: ['서울', '경기'],
  Team: {
    club: '1',
    clubname: '잔디FC',
  },
};

export const SIGN_UP_REQUEST = 'SIGN_UP_REQUEST';
export const SIGN_UP_SUCCESS = 'SIGN_UP_SUCCESS';
export const SIGN_UP_FAILURE = 'SIGN_UP_FAILURE';

export const LOG_IN_REQUEST = 'LOG_IN_REQUEST';
export const LOG_IN_SUCCESS = 'LOG_IN_SUCCESS';
export const LOG_IN_FAILURE = 'LOG_IN_FAILURE';

export const LOAD_USER_REQUEST = 'LOAD_USER_REQUEST';
export const LOAD_USER_SUCCESS = 'LOAD_USER_SUCCESS';
export const LOAD_USER_FAILURE = 'LOAD_USER_FAILURE';

export const LOG_OUT_REQUEST = 'LOG_OUT_REQUEST';
export const LOG_OUT_SUCCESS = 'LOG_OUT_SUCCESS';
export const LOG_OUT_FAILURE = 'LOG_OUT_FAILURE';

export const CHANGE_TO_REQUEST = 'CHANGE_TO_REQUEST';
export const CHANGE_TO_SUCCESS = 'CHANGE_TO_SUCCESS';
export const CHANGE_TO_FAILURE = 'CHANGE_TO_FAILURE';

export const PREEMPT_REQUEST = 'PREEMPT_REQUEST';
export const PREEMPT_SUCCESS = 'PREEMPT_SUCCESS';
export const PREEMPT_FAILURE = 'PREEMPT_FAILURE';

export default (state = initialState, action) => produce(state, (draft) => {
  switch (action.type) {
    case LOG_IN_REQUEST:
      draft.isLoggingIn = true;
      draft.logInErrorReason = null;
      break;
    case LOG_IN_SUCCESS:
      draft.isLoggingIn = false;
      draft.isLoggedIn = true;
      draft.me = dummyUser;
      break;
    case LOG_IN_FAILURE:
      draft.isLoggingIn = false;
      draft.isLoggedIn = false;
      draft.logInErrorReason = action.error;
      draft.me = null;
      break;
    case LOG_OUT_REQUEST:
      draft.isLoggedIn = false;
      draft.me = null;
      break;
    case SIGN_UP_REQUEST:
      draft.isSigningUp = true;
      draft.isSignedUp = false;
      draft.signUpErrorReason = null;
      break;
    case SIGN_UP_SUCCESS:
      draft.isSigningUp = false;
      draft.isSignedUp = true;
      break;
    case SIGN_UP_FAILURE:
      draft.isSigningUp = false;
      draft.signUpErrorReason = action.error;
      break;
    case CHANGE_TO_REQUEST:
      draft.isChangingTo = true;
      draft.isChangedTo = false;
      draft.changedToErrorReason = null;
      break;
    case CHANGE_TO_SUCCESS:
      draft.isChangingTo = false;
      draft.isChangedTo = true;
      break;
    case CHANGE_TO_FAILURE:
      draft.isChangingTo = false;
      draft.changedToErrorReason = action.error;
      break;
    case PREEMPT_REQUEST:
      draft.isPreempting = true;
      draft.isPreempted = false;
      draft.preemptErrorReason = null;
      break;
    case PREEMPT_SUCCESS:
      draft.isPreempting = false;
      draft.isPreempted = true;
      break;
    case PREEMPT_FAILURE:
      draft.isPreempted = false;
      draft.isPreempting = false;
      draft.preemptErrorReason = action.error;
      break;
    default:
      break;
  }
});
