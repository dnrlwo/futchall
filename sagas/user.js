import { all, delay, fork, put, takeEvery, takeLatest, call } from 'redux-saga/effects';
import axios from 'axios';
import {
  LOG_IN_REQUEST,
  LOG_IN_SUCCESS,
  LOG_IN_FAILURE,
  SIGN_UP_REQUEST,
  SIGN_UP_SUCCESS,
  SIGN_UP_FAILURE,
  CHANGE_TO_REQUEST,
  CHANGE_TO_SUCCESS,
  CHANGE_TO_FAILURE,
  LOG_OUT_REQUEST,
  LOG_OUT_FAILURE,
  LOG_OUT_SUCCESS,
} from '../reducers/user';

function loginAPI(data) {
  return axios.post('/user/login', data);
}

function* login(action) {
  try {
    const result = yield call(loginAPI, action.data);
    yield put({
      type: LOG_IN_SUCCESS,
      data: result.data,
    });
  } catch (e) {
    console.error(e);
    yield put({
      type: LOG_IN_FAILURE,
      error: e.response.data,
    });
  }
}

function* watchLogin() {
  yield takeEvery(LOG_IN_REQUEST, login);
}

function signUpAPI(data) {
  // 서버에 요청을 보내는 부분
  return axios.post('/user/signup', data);
}

function* signUp(action) {
  try {
    const result = yield call(signUpAPI, action.data);
    console.log(result);
    yield put({ // put은 dispatch 동일
      type: SIGN_UP_SUCCESS,
    });
  } catch (e) { // loginAPI 실패
    console.error(e);
    yield put({
      type: SIGN_UP_FAILURE,
      error: e.response.data,
    });
  }
}

function* watchSignUp() {
  yield takeEvery(SIGN_UP_REQUEST, signUp);
}

function changeToAPI(data) {
  return axios.post('/user/modify', data);
}
function* changeTo(action) {
  try {
    // yield call(changeToAPI(action));
    console.log('여기는 사가', action.data);
    yield delay(1000);
    yield put({
      type: CHANGE_TO_SUCCESS,
    });
  } catch (e) {
    yield put({
      type: CHANGE_TO_FAILURE,
      error: e.response.data,
    });
  }
}
function* watchChangeTo() {
  yield takeLatest(CHANGE_TO_REQUEST, changeTo);
}

function logOutAPI() {
  return axios.post('/user/logout');
}

function* logOut() {
  try {
    yield call(logOutAPI);
    yield put({
      type: LOG_OUT_SUCCESS,
    });
  } catch (err) {
    console.error(err);
    yield put({
      type: LOG_OUT_FAILURE,
      error: err.response.data,
    });
  }
}
function* watchLogOut() {
  yield takeLatest(LOG_OUT_REQUEST, logOut);
}

export default function* userSaga() {
  yield all([
    fork(watchLogin),
    fork(watchSignUp),
    fork(watchChangeTo),
    fork(watchLogOut),
  ]);
}
