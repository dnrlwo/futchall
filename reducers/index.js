import { combineReducers } from 'redux';
import { HYDRATE } from 'next-redux-wrapper';
import user from './user';
import location from './location';
import stadium from './stadium';
import team from './team';
import messenger from './messenger';

const rootReducer = combineReducers({
    index: (state={}, action) =>{
        switch (action.type) {
            case HYDRATE:
                console.log('HYDRATE', action);
                return {...state, ...action.payload};
            default:
                return state;
        }
    },
    user,
    location,
    stadium,
    team,
    messenger,
});

export default rootReducer;
