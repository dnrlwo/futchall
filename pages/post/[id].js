import React from 'react';
import { useRouter } from 'next/router';
import Head from 'next/head';
import axios from 'axios';
import { END } from 'redux-saga';
import { useSelector } from 'react-redux';
import { Col, Row } from 'antd';

import AppLayout2 from '../../components/AppLayout2';
import wrapper from '../../store/configureStore';
import { LOAD_MY_INFO_REQUEST } from '../../reducers/user';
import { SELECT_POST_REQUEST } from '../../reducers/post';
import PostComponent from '../../components/Post';

const Post = () => {
  const router = useRouter();
  const { id } = router.query;
  const { post, selectPostDone } = useSelector((state) => state.post);
  return (
    <AppLayout2>
      {
        selectPostDone && (
        <Head>
          <title>
            {post.User.nickname}의 게시글
          </title>
          <meta name="description" content={post.content} />
          <meta property="og:title" content={`${post.User.nickname}의 게시글`} />
          <meta property="og:description" content={post.content} />
          <meta property="og:image" content={post.Images[0] ? post.Images[0].src : 'https://futchall.com/favicon.ico'} />
          <meta property="og:url" content={`https://futchall.com/post/${id}`} />
        </Head>
        )
      }
      <Row style={{ paddingTop: '10px' }}>
        <Col xs={{ span: 24 }} sm={{ span: 24 }} md={{ span: 12, offset: 6 }}>
          <PostComponent data={post} key={post.id} />
        </Col>
      </Row>
    </AppLayout2>
  );
};

export const getServerSideProps = wrapper.getServerSideProps(async (context) => {
  const cookie = context.req ? context.req.headers.cookie : '';
  axios.defaults.headers.common.Authorization = '';
  let token = '';
  if (context.req && cookie) {
    if (cookie.indexOf(';') !== -1) {
      const index = cookie.indexOf('RefreshToken');
      token = cookie.slice(index + 13, cookie.indexOf(';', index));
    } else {
      token = cookie.slice(13);
    }
    if (token) {
      axios.defaults.headers.common.Authorization = `Bearer ${token}`;
      context.store.dispatch({ type: LOAD_MY_INFO_REQUEST });
    }
  }
  context.store.dispatch({
    type: SELECT_POST_REQUEST,
    data: {
      id: context.params.id,
    },
  });
  context.store.dispatch(END);
  await context.store.sagaTask.toPromise();
});

export default Post;
