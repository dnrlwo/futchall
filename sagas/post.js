import axios from 'axios';
import shortId from 'shortid';
import { all, fork, put, call, takeLatest, throttle } from 'redux-saga/effects';

import {
  ADD_COMMENT_FAILURE,
  ADD_COMMENT_REQUEST,
  ADD_COMMENT_SUCCESS,
  ADD_POST_FAILURE,
  ADD_POST_REQUEST,
  ADD_POST_SUCCESS,
  LOAD_POSTS_FAILURE,
  LOAD_POSTS_REQUEST,
  LOAD_POSTS_SUCCESS,
  REMOVE_POST_FAILURE,
  REMOVE_POST_REQUEST,
  REMOVE_POST_SUCCESS,
  UPLOAD_IMAGES_REQUEST,
  UPLOAD_IMAGES_SUCCESS,
  UPLOAD_IMAGES_FAILURE,
  SELECT_POST_REQUEST,
  SELECT_POST_SUCCESS,
  SELECT_POST_FAILURE
} from '../reducers/post';
import { ADD_POST_TO_ME, REMOVE_POST_OF_ME } from '../reducers/user';

function loadPostsAPI(data) {
  return axios.get(`/posts?where=${data.where}&id=${data.id}&lastId=${data.lastId || 0}`);
}

function* loadPosts(action) {
  try {
    const result = yield call(loadPostsAPI, action.data);
    yield put({
      type: LOAD_POSTS_SUCCESS,
      data: result.data,
    });
  } catch (err) {
    console.error(err);
    yield put({
      type: LOAD_POSTS_FAILURE,
      data: err.response.data,
    });
  }
}

function addPostAPI(action) {
  if (data.where === 'team') {
    return axios.post('/post/team', action.data, { headers: { Authorization: `Bearer ${action.token}` } });
  }
  return axios.post('/post/stadium', action.data, { headers: { Authorization: `Bearer ${action.token}` } });
}

function* addPost(action) {
  try {
    const result = yield call(addPostAPI, action);
    yield put({
      type: ADD_POST_SUCCESS,
      data: result.data,
    });
    yield put({
      type: ADD_POST_TO_ME,
      data: result.data.id,
    });
  } catch (err) {
    console.error(err);
    yield put({
      type: ADD_POST_FAILURE,
      data: err.response.data,
    });
  }
}

function removePostAPI(action) {
  return axios.delete(`/post/${action.data.id}`, { headers: { Authorization: `Bearer ${action.token}` } });
}

function* removePost(action) {
  try {
    const result = yield call(removePostAPI, action);
    yield put({
      type: REMOVE_POST_SUCCESS,
      data: result.data.PostId,
    });
    yield put({
      type: REMOVE_POST_OF_ME,
      data: result.data.PostId,
    });
  } catch (err) {
    console.error(err);
    yield put({
      type: REMOVE_POST_FAILURE,
      data: err.response.data,
    });
  }
}

function addCommentAPI(action) {
  return axios.post(`/post/${action.data.postId}/comment`, action.data, { headers: { Authorization: `Bearer ${action.token}` }});
}

function* addComment(action) {
  try {
    const result = yield call(addCommentAPI, action);
    yield put({
      type: ADD_COMMENT_SUCCESS,
      data: result.data,
    });
  } catch (err) {
    yield put({
      type: ADD_COMMENT_FAILURE,
      data: err.response.data,
    });
  }
}

function selectPostAPI(data) {
  return axios.get(`/post/${data.id}`);
}

function* selectPost(action) {
  try {
    const result = yield call(selectPostAPI, action.data);
    yield put({
      type: SELECT_POST_SUCCESS,
      data: result.data,
    });
  } catch (err) {
    console.error(err);
    yield put({
      type: SELECT_POST_FAILURE,
      error: err.response.data,
    });
  }
}

function* watchSelectPost() {
  yield takeLatest(SELECT_POST_REQUEST, selectPost);
}

function uploadImagesAPI(data) {
  return axios.post('/post/images', data);
}

function* uploadImages(action) {
  try {
    const result = yield call(uploadImagesAPI, action.data);
    yield put({
      type: UPLOAD_IMAGES_SUCCESS,
      data: result.data,
    });
  } catch (err) {
    console.error(err);
    yield put({
      type: UPLOAD_IMAGES_FAILURE,
      error: err.response.data,
    });
  }
}

function* watchUploadImages() {
  yield takeLatest(UPLOAD_IMAGES_REQUEST, uploadImages);
}

function* watchLoadPosts() {
  yield throttle(5000, LOAD_POSTS_REQUEST, loadPosts);
}

function* watchAddPost() {
  yield takeLatest(ADD_POST_REQUEST, addPost);
}

function* watchRemovePost() {
  yield takeLatest(REMOVE_POST_REQUEST, removePost);
}

function* watchAddComment() {
  yield takeLatest(ADD_COMMENT_REQUEST, addComment);
}

export default function* postSaga() {
  yield all([
    fork(watchAddPost),
    fork(watchLoadPosts),
    fork(watchRemovePost),
    fork(watchAddComment),
    fork(watchUploadImages),
    fork(watchSelectPost),
  ]);
}
