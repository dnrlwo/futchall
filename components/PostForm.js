import React, { useCallback, useState, useEffect } from 'react';
import { useSelector, useDispatch, shallowEqual } from 'react-redux';
import PropTypes from 'prop-types';
import { Form, Avatar, Input, Button, Divider, Space, Tag, Upload, Modal } from 'antd';
import styled from 'styled-components';
import { FileImageOutlined, CalendarOutlined, PlusOutlined } from '@ant-design/icons';

import ReservationMatch from './ReservationMatch';
import { ADD_POST_REQUEST } from '../reducers/post';
import imageUploader from '../util/imageUploader';
import getBase64 from '../util/getBase64';


const PostFormDiv = styled.div`
  border-radius: 15px;
  padding: 10px;
  border: 1px solid #f0f0f0;
  height: auto;
  margin-bottom: 10px;
  background-color: #fff;
`;
const PostForm = ({ where, req }) => {
  // const { where, req } = props;
  const dispatch = useDispatch();
  const [form] = Form.useForm();
  const { me } = useSelector((state) => state.user, shallowEqual);
  const { addPostLoading, addPostDone } = useSelector((state) => state.post, shallowEqual);
  const [visible, setVisible] = useState(false);
  const [enrollment, setEnrollment] = useState(false);
  const [matchInfo, setMatchInfo] = useState({ stadiumTitle: null, stadiumReq: null, date: null });
  const [previewVisible, setPreviewVisible] = useState(false);
  const [uploadImage, setUploadImage] = useState(false);
  const [previewImage, setPreviewImage] = useState('');
  const [imageList, setImagelist] = useState([]);
  const [dbImage, setDbImage] = useState([]);

  const onSubmit = useCallback(() => {
    dispatch({
      type: ADD_POST_REQUEST,
      data: {
        content: form.getFieldValue('content'),
        where,
        req,
        matchInfo,
        image: dbImage,
      },
    });
  }, [matchInfo, dbImage]);

  const onAdjustMatch = useCallback(() => {
    setVisible(true);
  }, []);
  const handlePreview = async (file) => {
    if (!file.url && !file.preview) {
      file.preview = await getBase64(file.originFileObj);
    }
    setPreviewImage(file.url || file.preview);
    setPreviewVisible(true);
  };
  useEffect(() => {
    if (addPostDone) {
      form.resetFields(['content']);
      setMatchInfo({
        stadiumTitle: null,
        date: null,
      });
      setUploadImage(false);
      setDbImage([]);
      setImagelist([]);
    }
  }, [addPostDone]);
  return (
    <PostFormDiv>
      <Form
        form={form}
        hideRequiredMark
        encType="multipart/form-data"
        onFinish={onSubmit}
      >
        <Form.Item
          label={<Avatar shape="circle">{me?.nickname}</Avatar>}
          colon={false}
          name="content"
          required
          // rules={[{ required: trued, message: '무슨 말을 하고 싶으신가요?' }]}
        >
          <Input.TextArea
            autoSize={{ minRows: 2, maxRows: 10 }}
            style={{ border: '1px solid #f0f0f0', borderRadius: '15px', backgroundColor: '#fafafa', color: '#000000d9' }}
            placeholder="무슨 생각을 하고 계신가요?"
          />
        </Form.Item>
        { uploadImage
          && (
          <Form.Item>
            <Upload
              listType="picture-card"
              action={(file) => imageUploader('http://localhost:3065/post/images', file).then((response) => setDbImage(dbImage.concat(response.data[0])))}
              onChange={({ fileList }) => setImagelist(fileList)}
              fileList={imageList}
              onPreview={(file) => handlePreview(file)}
              onRemove={(file) => {
                let index = -1;
                setImagelist(imageList.filter((v, i) => {
                  if (v.uid !== file.uid) {
                    return true;
                  }
                  index = i;
                  return false;
                }));
                setDbImage(dbImage.filter((v, i) => i !== index));
              }}
            >
              <div>
                <PlusOutlined />
                <div style={{ marginTop: 8 }}>Upload</div>
              </div>
            </Upload>
            <Modal
              visible={previewVisible}
              footer={null}
              onCancel={() => setPreviewVisible(false)}
            >
              <img alt="example" style={{ width: '100%' }} src={previewImage} />
            </Modal>
          </Form.Item>
          )}
        {
          enrollment && (
          <div>
            <span>경기 일정</span>
            <Tag
              closable
              onClose={() => { setEnrollment(false); }}
              color="#1890ff"
            >
              {matchInfo.stadiumTitle + ' ' + matchInfo.date}
            </Tag>
          </div>
          )
        }
        <Divider />
        <Form.Item
          style={{ marginBottom: 0, textAlign: 'right' }}
        >
          <Space>
            <Button type="default" htmlType="button" shape="round" onClick={() => setUploadImage(true)}><FileImageOutlined />사진</Button>
            {!enrollment && <Button type="default" htmlType="button" shape="round" onClick={onAdjustMatch}><CalendarOutlined />경기 일정</Button>}
            <Button type="primary" htmlType="submit" shape="round" loading={addPostLoading}>게시하기</Button>
          </Space>
        </Form.Item>
      </Form>
      {visible && <ReservationMatch visible={visible} setVisible={setVisible} onLoadPost={setMatchInfo} stadiumReq={where === 'stadium' && req} setEnrollment={setEnrollment}/>}
    </PostFormDiv>
  );
};

PostForm.propTypes = {
  where: PropTypes.string.isRequired,
  req: PropTypes.string.isRequired,
};

export default PostForm;
