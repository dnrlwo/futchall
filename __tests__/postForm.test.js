import React from 'react';
import '../matchMedia.mock';
import { mount, shallow } from 'enzyme';
import nock from 'nock';
import configureMockStore from 'redux-mock-store'

import { Provider } from 'react-redux';
import {configureStore} from '../store/configureStore'
import PostForm from '../components/PostForm';
import { LOAD_MY_INFO_SUCCESS, SET_MY_TOKEN } from '../reducers/user';
import { ADD_POST_SUCCESS } from '../reducers/post';
import { Input } from 'antd';

describe('Post', () => {
  const store = configureStore();
  let component = null;
  const props = {
    where: "team",
    req: "1"
  };
  it('Mock Store', async() => {
    await store.dispatch({
      type: SET_MY_TOKEN,
      data: 'aaa.bbb.ccc'
    });
    await store.dispatch({
      type: LOAD_MY_INFO_SUCCESS,
      data: {
        Images: [],
        nickname: '우기',
      }
    });
  });

  it('renders postform', () => {
    component = mount(
      <Provider store={store}>
        <PostForm {...props} />
      </Provider>
    );
  });
  
  it('renders map', () => {
    component.find('textarea').simulate('change', {
      target: { value: 'testing' }
    });
    expect(component.find('textarea').props().value).toEqual('testing');
    component.simulate('submit');
    store.dispatch({
      type: ADD_POST_SUCCESS,
      data: {
        User: {
          id: 1,
          nickname: '우기'
        },
        content: "안녕"
      }
    });
    component.setProps();
    // component.update();
    expect(component.find('textarea').props().value.length).toBe(0);
  })
});