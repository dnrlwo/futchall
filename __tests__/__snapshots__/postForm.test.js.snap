// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post renders postform 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "sagaTask": Object {
          "@@redux-saga/TASK": true,
          "cancel": [Function],
          "cont": [Function],
          "context": Object {},
          "end": [Function],
          "error": [Function],
          "id": 1,
          "isAborted": [Function],
          "isCancelled": [Function],
          "isRoot": true,
          "isRunning": [Function],
          "joiners": Array [],
          "meta": Object {
            "location": undefined,
            "name": "rootSaga",
          },
          "queue": Object {
            "abort": [Function],
            "addTask": [Function],
            "cancelAll": [Function],
            "getTasks": [Function],
          },
          "result": [Function],
          "setContext": [Function],
          "toPromise": [Function],
        },
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "sagaTask": Object {
            "@@redux-saga/TASK": true,
            "cancel": [Function],
            "cont": [Function],
            "context": Object {},
            "end": [Function],
            "error": [Function],
            "id": 1,
            "isAborted": [Function],
            "isCancelled": [Function],
            "isRoot": true,
            "isRunning": [Function],
            "joiners": Array [],
            "meta": Object {
              "location": undefined,
              "name": "rootSaga",
            },
            "queue": Object {
              "abort": [Function],
              "addTask": [Function],
              "cancelAll": [Function],
              "getTasks": [Function],
            },
            "result": [Function],
            "setContext": [Function],
            "toPromise": [Function],
          },
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <PostForm
    req="1"
    where="team"
  />
</ContextProvider>
`;

exports[`Post store snapshot 1`] = `
Object {
  "dispatch": [Function],
  "getState": [Function],
  "replaceReducer": [Function],
  "sagaTask": Object {
    "@@redux-saga/TASK": true,
    "cancel": [Function],
    "cont": [Function],
    "context": Object {},
    "end": [Function],
    "error": [Function],
    "id": 1,
    "isAborted": [Function],
    "isCancelled": [Function],
    "isRoot": true,
    "isRunning": [Function],
    "joiners": Array [],
    "meta": Object {
      "location": undefined,
      "name": "rootSaga",
    },
    "queue": Object {
      "abort": [Function],
      "addTask": [Function],
      "cancelAll": [Function],
      "getTasks": [Function],
    },
    "result": [Function],
    "setContext": [Function],
    "toPromise": [Function],
  },
  "subscribe": [Function],
  Symbol(observable): [Function],
}
`;
